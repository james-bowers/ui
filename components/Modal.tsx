import * as React from "react";
import styled, { keyframes } from "./styled";

type ModalProps = {
  children: any;
  open: boolean;
  onWrapperClicked?: Function;
};

let ModalWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(255, 255, 255, 0.5);
`;

let ModalContent = styled.div`
  max-width: 50%;
  max-height: 50%;
  background: ${props => props.theme.primaryColor};
  color: ${props => props.theme.primaryColorInverted};
`;

export default (props: ModalProps) => (
  <div>
    {props.open && (
      <ModalWrapper
        onClick={() => props.onWrapperClicked && props.onWrapperClicked()}
      >
        <ModalContent
          onClick={event => {
            event.preventDefault();
            event.stopPropagation();
            return false;
          }}
        >
          {props.children}
        </ModalContent>
      </ModalWrapper>
    )}
  </div>
);
