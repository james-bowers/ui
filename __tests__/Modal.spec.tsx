import * as React from "react";
import { configure, shallow } from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

import Modal from "../components/Modal";

describe("<Modal />", () => {
  test("shows the modal", () => {
    const tree = shallow(<Modal open={true}>content inside modal</Modal>);
    expect(toJson(tree)).toMatchSnapshot();
  });

  test("hides the modal", () => {
    const tree = shallow(<Modal open={false}>content inside modal</Modal>);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
