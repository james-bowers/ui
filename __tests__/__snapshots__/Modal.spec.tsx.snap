// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> hides the modal 1`] = `<div />`;

exports[`<Modal /> shows the modal 1`] = `
<div>
  <styled.div
    onClick={[Function]}
  >
    <styled.div
      onClick={[Function]}
    >
      content inside modal
    </styled.div>
  </styled.div>
</div>
`;
